modules/RTC/CodecSelection.spec.js

```diff
@@ -0,0 +1,214 @@
+    let codecSelection;
+    let conference;
+    let connection;
+    let jingleSession;
+    let options;
+    let participant1, participant2;
+    let rtc;
+    const SID = 'sid12345';
+
+    beforeEach(() => {
+        conference = new MockConference();
+        connection = new MockStropheConnection();
+        jingleSession = new JingleSessionPC(
+            SID,
+            'peer1',
+            'peer2',
+            connection,
+            { },
+            { },
+            false,
+            false);
+
+        rtc = new MockRTC();
+
+        jingleSession.initialize(
+            /* ChatRoom */ new MockChatRoom(),
+            /* RTC */ rtc,
+            /* Signaling layer */ conference._signalingLayer,
+            /* options */ { });
+        conference.jvbJingleSession = jingleSession;
+    });
+
+    describe('when codec preference list is used in config.js', () => {
+        beforeEach(() => {
+            options = {
+                jvb: {
+                    preferenceOrder: [ 'VP9', 'VP8', 'H264' ]
+                }
+            };
+
+            codecSelection = new CodecSelection(conference, options);
+            spyOn(jingleSession, 'setVideoCodecs');
+        });
+
+        it('and remote endpoints use the new codec selection logic', () => {
+            // Add a second user joining the call.
+            participant1 = new MockParticipant('remote-1');
+            conference.addParticipant(participant1, [ 'vp9', 'vp8' ]);
+
+            expect(jingleSession.setVideoCodecs).toHaveBeenCalledWith([ 'vp9', 'vp8' ]);
+
+            // Add a third user joining the call with a subset of codecs.
+            participant2 = new MockParticipant('remote-2');
+            conference.addParticipant(participant2, [ 'vp8' ]);
+
+            expect(jingleSession.setVideoCodecs).toHaveBeenCalledWith([ 'vp8' ]);
+
+            // Make p2 leave the call
+            conference.removeParticipant(participant2);
+            expect(jingleSession.setVideoCodecs).toHaveBeenCalledWith([ 'vp9', 'vp8' ]);
+        });
+
+        it('and remote endpoints use the old codec selection logic (RN)', () => {
+            // Add a second user joining the call.
+            participant1 = new MockParticipant('remote-1');
+            conference.addParticipant(participant1, null, 'vp8');
+
+            expect(jingleSession.setVideoCodecs).toHaveBeenCalledWith([ 'vp8' ]);
+
+            // Add a third user (newer) to the call.
+            participant2 = new MockParticipant('remote-2');
+            conference.addParticipant(participant2, [ 'vp9', 'vp8' ]);
+
+            expect(jingleSession.setVideoCodecs).toHaveBeenCalledWith([ 'vp8' ]);
+
+            // Make p1 leave the call
+            conference.removeParticipant(participant1);
+            expect(jingleSession.setVideoCodecs).toHaveBeenCalledWith([ 'vp9', 'vp8' ]);
+        });
+    });
+
+    describe('when deprecated configs are used in config.js', () => {
```

Comment:
Can you add a test which tests the opposite prepferences? Web prefers vp9 and disabled h264 and mobile the opposite. It should result in vp8.